pinot-core/src/main/java/org/apache/pinot/core/auth/BasicAuthPrincipal.java

```diff
@@ -50,6 +50,6 @@ public boolean hasTable(String tableName) {
   }
 
   public boolean hasPermission(String permission) {
-    return _permissions.isEmpty() || _permissions.contains(permission);
+    return _permissions.isEmpty() || _permissions.stream().anyMatch(d -> d.equalsIgnoreCase(permission));
```

Comment:
This can be slow. Suggest fix the permissions to lower case and do `_permissions.contains(permission.toLowerCase())`

Reply:
done and added unit tests.